R/clean_glm_data.R

```diff
@@ -0,0 +1,27 @@
+clean_glm_data <- function(mod, data, weights){
+
+  # Grab only necessary variables
+  data <- get_all_vars(mod, data)
+
+  if(!is.null(weights) & length(weights) != nrow(data))
+    stop('`weights` and `data` must be the same length.')
+
+  # Add weights
```

Comment:
Alternative approach is 

* Remove the `weights` and the `data` arguments from `marg` entirely
* add a new `subset` argument that expects a logical vector
* if `subset` is `NULL` then subset the data based on the names of `mod$prior.weights`
* if `subset` isn't `NULL`, subset the data based on the intersection of the names of `mod$prior.weights` and the `subset` vector?